src/js/actions/authActions.js

```diff
@@ -0,0 +1,85 @@
+import fetch from 'fetch-everywhere';
+import {push} from 'react-router-redux';
+import urlFor from '../helpers/urlFor';
+
+const requestLogout = () => ({
+  type: 'LOGOUT_REQUEST',
+  isFetching: true,
+  isAuthenticated: true
+});
+
+const receiveLogout = () => ({
+  type: 'LOGOUT_SUCCESS',
+  isFetching: false,
+  isAuthenticated: false
+});
+
+// Logs the user out
+export const logoutUser = () => (dispatch) => {
+  dispatch(requestLogout());
+  localStorage.removeItem('idToken');
+  localStorage.removeItem('store');
+  dispatch(receiveLogout());
+  dispatch(push('/'));
+};
+
+const requestLogin = (code) => ({
+  type: 'LOGIN_REQUEST',
+  isFetching: true,
+  isAuthenticated: false,
+  code
+});
+
+const receiveLogin = (token) => ({
+  type: 'LOGIN_SUCCESS',
+  isFetching: false,
+  isAuthenticated: true,
+  user: token
+});
+
+const loginError = (err) => ({
+  type: 'LOGIN_FAILURE',
+  isFetching: false,
+  isAuthenticated: false,
+  err
+});
+
+export const loginUser = (code) => (dispatch) => {
+  // API Route for Slack Authentication
+  const url = urlFor('/api/v1/auth/slack');
```

Comment:
Nah.  When writing ES6 convention is to ALWAYS use `const` or `let` and never use `var` (there is weird scoping with var that most people don't understand).  You use `const` for variables that don't change.  You use `let` for variables that do.

`let` is better than var because:

```
"use strict";

for(var i =0; i < 10; i++) {
  console.log("Hello " + i);
}

console.log("The value of i is: " + i);
```

`var` leaks the variable out and it's active.  `let` doesn't.  

It's more like scala or swift.